File: data-structures/arrays/javascript/arrayOperations.js
/**
 * Declaration
 */
const arr = []; // empty array
const arrWithValues = [1, 2, 3, 4, 5]; // Array  with values
const arrWithSize = new Array(5); // Array with size
const arrWithSizePrefilled  = new Array(100).fill(-1);

/**
 * Accessing elements
 */
let firstElement = arrWithValues[0]; // 1

/**
 * Setting elements
 */
arrWithValues[0] = 100;

/**
 * Array length
 */
const length = arrWithValues.length;

/**
 * Iterating over arrays
 */
for (let value of arrWithValues) {
  console.log(value);
}
for (let i = 0; i < arrWithValues.length; i++) {
  console.log(arrWithValues[i]);
}

/**
 * Multi-dimensional arrays
 */
const multiArray = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
];
console.log(multiArray[1][2]); // 6

/**
 * Common methods
 */
// add to end
arrWithValues.push(200);

// remove from end
arrWithValues.pop();

// add to beginning
arrWithValues.unshift(10);

// remove from beginning
let firstItem = arrWithValues.shift();

// slice array - creates new array object not including end
let start = 0;
let end = 3;
const newArr = arrWithValues.slice(start, end);
console.log(arrWithValues);
console.log(newArr);

// splice array - add, remove or replace existing elements in place
const months = ["Jan", "Mar", "Apr", "Jun"];
console.log(months);
let startIndex = 1;
let numOfElementsToRemove = 0;
months.splice(startIndex, numOfElementsToRemove, "Feb");
console.log(months)

File: data-structures/arrays/java/ArrayOperations.java
public class ArrayOperations {
    public static void main(String[] args){
        // array declaration
       int[] arr;
       // array of size 5 with default values (0 for int)
       arr =  new int[5];
       // array with values
       int[] arrWithValues = {1, 2, 3, 4, 5};
       
       // 0 0 0 0 0
       for (int val: arr) {
        System.out.print(val + " ");
       }
       System.out.println();

       // 1 2 3 4 5
       for (int val: arrWithValues){
        System.out.print(val + " ");
       }
       System.out.println();

       // accessing elements
       int lastElement = arrWithValues[4];
       System.out.println(lastElement);

       // Setting elements
       arrWithValues[0] = 100;
       // 100 2 3 4 5
       for (int val: arrWithValues){
        System.out.print(val + " ");
       }
       System.out.println();

       // array length
       int length = arrWithValues.length;
       // 5
       System.out.println(length);

       // iterating over elements 
       for (int i  = 0; i < arrWithValues.length; i++){
        System.out.print(arrWithValues[i] + " ");
       }
       System.out.println();

       // iterating over elements with enhanced for loop
       for (int value: arrWithValues){
        System.out.print(value + " ");
       }
       System.out.println();

       // ---
      // multi-dimensional arrays
      int [][] multiArray = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
      //6
      System.out.println(multiArray[1][2]);
    }
}